### ISSUE TITLE
[VSLCV] Bar chart design enhancements

### ISSUE THREAD

**Issue by srallen** [zooniverse/front-end-monorepo#1440]
**Package** lib-classifier

**Describe the solution you'd like**

After meeting with the researchers, we have to balance design and accessibility needs with the needs of the project, so there are a few discussion points we need to review with regard to the bar chart and how it is displayed in the composite VSLCV:

- The specific values of the bars in their case shouldn't distract volunteers from making variable star classifications. They're worried that some volunteers may get too focus on specific numerical values and that might lead them to make inaccurate classifications. To this end, they want the axis values to be hidden. What they want volunteers to do is to just compare the bar height size of the red light and blue light spectra to each other for period and amplitude.
- However, generally speaking, we need labels for accessibility. They've recommended just having labels like 'tall' and 'short' be read out. I'm not sure if that's viable because how do you determine what is 'tall' and 'short' as it's inherently an arbitrary distinction.
  - We could choose to allow project builders to submit aria labels in the JSON, but I'd rather not go this route as it may open up the bar chart to more complexity and potentially poor labeling.
  - I showed the tooltips from #1375, and that was an acceptable middle ground for displaying specific values or labels. If the axes are hidden, though, are the tooltips discoverable?

So the question is: How can we make the display of the axes and specific individual bar values configurable and also maintain accessibility? @eatyourgreens do you have any suggestions for this?

Additional TODO:
- [ ] How the labels are read out by the bar chart currently still need adjusting [comment](https://github.com/zooniverse/front-end-monorepo/pull/1375#pullrequestreview-345681695).

**Comment by eatyourgreens** [comment]
I’m not really understanding the problem. When you listen to the axes in VoiceOver, you hear a bunch of annoying groups, each consisting of one number. That’s because each individual label is implemented as a group by VX.

It’s hard to comment without having a real example to refer to.

**Comment by eatyourgreens** [comment]
For a blind person, all you need are the bars to be read out. Each bar announces its x and y value. Surely the axes are irrelevant? If not, the daily stats chart needs rethinking too.

**Comment by srallen** [comment]
The issue is that they don't want to use the actual x and y values for the labels generally speaking because they think it's a distraction from the actual classification task. They're ok with the tooltips, but I'm not sure if those are discoverable enough if the axes are hidden from the display.

**Comment by srallen** [comment]
Without at least individual bar labels, I'm concerned with having colors for each bar that isn't distinguishable enough for color blindness too.

**Comment by beckyrother** [comment]
Seems like the most simple route will be the best here: Using the tooltips that show the values on mouseover so they're not always visible/distracting, but will be the most helpful for screenreaders.

**Comment by eatyourgreens** [comment]
>The issue is that they don't want to use the actual x and y values for the labels generally speaking because they think it's a distraction from the actual classification task. They're ok with the tooltips, but I'm not sure if those are discoverable enough if the axes are hidden from the display. I realize this is a visual UI issue, not screen reader issue.

RIght, I understand it now. My initial reaction is that this is ridiculous. How is the chart conveying information if there are no axis labels?

>They've recommended just having labels like 'tall' and 'short' be read out.

This is an absolutely terrible suggestion, not just from an accessibility point of view but because this could be classified by a machine with no human intervention.

>What they want volunteers to do is to just compare the bar height size of the red light and blue light spectra to each other for period and amplitude.

Why does this need human classifiers?

>How the labels are read out by the bar chart currently still need adjusting

I think they're awkward in VoiceOver because of the VX implementation. VO is reading out all the SVG group elements individually. My suggestion would be to wrap the VX Axis in something that announces the whole thing as image. For example:
```jsx
<g
  role="image"
  aria-label="Y-axis: Luminosity from 0.1 to 1.2"
>
  // Axis elements go here but should be hidden from a screen reader by the parent image role
</g>
```
It seems to me that the `aria-label` could be generated from the axis label, minimum value and maximum value. What do you think?

**Comment by eatyourgreens** [comment]
>Using the tooltips that show the values on mouseover so they're not always visible/distracting

I hate to keep repeating this but mouseover/hover is not available as an interaction for touch interfaces, voice control, keyboard control or screen readers.

**Comment by srallen** [comment]
Just for reference, the bar chart will appear in a composite with an image and two scatter plots ((design)[https://projects.invisionapp.com/d/main?origin=v7#/projects/prototypes/17294981]). The classification task is going to be categorization of variable stars. Alone, yes, I would agree and absolutely say no to this request, but I think we can come to some kind of compromise based for the request in the composite. 

At a minimum we need labels for each bar because the color has meaning. Perhaps just hiding the ticks on the axis will be sufficient. 

Fixing how the axes read out is a slight addendum that I lumped into this here, but isn't the main design issue. Perhaps that was a mistake in my issue organizing. 

> I hate to keep repeating this but mouseover/hover is not available as an interaction for touch interfaces, voice control, keyboard control or screen readers.

I think Becky does understand this. Thanks for the reminder though.

**Comment by beckyrother** [comment]
Unless I'm missing something, it seems like the only labels that would really be useful would be 'red light' and 'blue light', which are visually indicated on the key and won't need to be reiterated: 
![Screenshot_1_30_20__2_33_PM](https://user-images.githubusercontent.com/1392130/73487910-84feaa80-436d-11ea-981d-2f88a8093559.png)

Tick marks on the chart without any context would be a worse experience than nothing at all – it might be perceived as an error or accidental omission, which would be distracting at best and at worst decrease a volunteer's faith in the data.

**Comment by srallen** [comment]
Researcher request:

> Right now the period bar chart does not change, but the axis labels do when switching periods. The numbers on that axis are quite small. The way we got around this in our mock up was the fix the range on the bar (i.e. for every chart period goes from 0 ---> 10). Then changing the period would change the length of the bar instead of the axis. I'm guessing (but this is hardly my expertise) that this would also be better for accessibility as it would not require the user to read off very tiny numbers. Is this possible?

**Comment by beckyrother** [comment]
I still agree with Past Becky – charts axes need labels whether or not the axis changes based on the period. The numbers are 10px, which is small but still a fairly reasonable size

**Comment by srallen** [comment]
Yes, I agree, the labels should remain. I think we can at least have a look at fixing the range of the axes as well as possibly add an animation in to make it more clear when the period bar chart changes.
